storage/database/db_manager.go

```diff
@@ -1057,30 +1153,106 @@ func (dbm *databaseManager) WriteMerkleProof(key, value []byte) {
 
 // Cached Trie Node operation.
 func (dbm *databaseManager) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+	if dbm.inMigration {
```

Comment:
Why don't you separate Read function in migration and Read function not in migration?

Reply:
If they are separated, the caller of read-request function should know whether the node is in migration process or not and `if-else` statements are needed for every functions. Instead, this way doesn't change the way it is used for now and only the migrator will use the newly added functions with the suffix `...FromOld`